packages/inter-protocol/src/vaultFactory/proceeds.js

```diff
@@ -0,0 +1,273 @@
+      const lessCollat = AmountMath.add(debtCollat, penaltyCollat);
+
+      const maxCollat = subtractToEmpty(vCollat, lessCollat);
+      if (!AmountMath.isEmpty(leftToStage)) {
+        const collatReturn = AmountMath.min(leftToStage, maxCollat);
+        leftToStage = AmountMath.subtract(leftToStage, collatReturn);
+        plan.transfersToVault.push([vaultIndex, { Collateral: collatReturn }]);
+      }
+    }
+
+    const hasCollateralToDistribute = !AmountMath.isEmpty(
+      distributableCollateral,
+    );
+    plan.collateralForReserve = hasCollateralToDistribute
+      ? AmountMath.add(leftToStage, totalPenalty)
+      : collateralProceeds;
+  };
+
+  const runFlow2a = () => {
+    // charge penalty if proceeds are sufficient
+    const penaltyInMinted = ceilMultiplyBy(totalDebt, penaltyRate);
+    const recoveredDebt = AmountMath.min(
+      AmountMath.add(totalDebt, penaltyInMinted),
+      mintedProceeds,
+    );
+
+    plan.debtToBurn = recoveredDebt;
+
+    const distributable = subtractToEmpty(mintedProceeds, recoveredDebt);
+    let mintedRemaining = distributable;
+
+    const vaultPortion = makeRatioFromAmounts(distributable, totalCollateral);
+
+    // iterate from best to worst returning remaining funds to vaults
+    for (const [vaultIndex, balances] of vaultsBalances.entries()) {
+      // from best to worst, return minted above penalty if any remains
+      const vaultShare = floorMultiplyBy(balances.collateral, vaultPortion);
+      updatePhantomDebt(balances);
+
+      if (!AmountMath.isEmpty(mintedRemaining)) {
+        const mintedToReturn = AmountMath.isGTE(mintedRemaining, vaultShare)
+          ? vaultShare
+          : mintedRemaining;
+        mintedRemaining = AmountMath.subtract(mintedRemaining, mintedToReturn);
+        plan.transfersToVault.push([vaultIndex, { Minted: mintedToReturn }]);
+      }
+    }
+  };
+
+  const runFlow2b = () => {
+    plan.debtToBurn = totalDebt;
+    plan.mintedForReserve = accounting.overage;
+
+    // reconstitute vaults until collateral is insufficient
+    let reconstituteVaults = AmountMath.isGTE(collateralProceeds, totalPenalty);
+
+    // charge penalty if proceeds are sufficient
+    const distributableCollateral = subtractToEmpty(
+      collateralProceeds,
+      totalPenalty,
+    );
+
+    plan.collatRemaining = distributableCollateral;
+
+    let shortfallToReserve = accounting.shortfall;
+    const reduceCollateral = amount =>
+      (plan.actualCollateralSold = AmountMath.add(
+        plan.actualCollateralSold,
+        amount,
+      ));
+
+    // iterate from best to worst attempting to reconstitute, by
+    // returning remaining funds to vaults
+    for (const [vaultIndex, balances] of vaultsBalances.entries()) {
+      const { collateral: vCollat, presaleDebt } = balances;
+
+      // according to #7123, Collateral for penalty =
+      //    vault debt / total debt * total liquidation penalty
```

Comment:
When I'm working through the math, this is immensely easier to read.  Ratio math requires that the numerator be a different brand from the denominator, so we calculate with vault debt * (total liquidation penalty / total debt).
```suggestion
      //    total liquidation penalty * vault debt / total debt
```

Reply:
done